bolt-design-system/bolt
# fix: Add simple button for forms

## Jira

http://vjira2:8080/browse/WWWD-2739

See also
http://vjira2:8080/browse/WWWD-2724
Possibly http://vjira2:8080/browse/WWWD-2711

## Summary

Adds a variant of button for use in forms that does not invoke any Bolt javascript

## Details

The Preact-based `<bolt-button>` component has a history of being obstinate when it comes to Drupal forms.  When we were initially building out forms in Bolt, there were several obstacles and revisions, but we were ultimately able to get them resolved such that Drupal forms worked as expected when using `<bolt-button>`.

With Bolt 2.0, however, there were severe regressions to forms (see linked Jira issues).  These were:
- painful for stakeholders (Ross and Liz), who got chewed out because forms weren't working as expected.
- painful for myself, @charginghawk, @krlucas, and others who spent significant development and/or communication time last week trying to resolve these elusive and high priority issues.
- never fully resolved.  For example, http://vjira2:8080/browse/WWWD-2739 was ultimately resolved by disabling AJAX form submission. 

Is Drupal doing something "wrong" here when it comes to attaching javascript behaviors to buttons?  Maybe.  Maybe not.  We've sunk enough time troubleshooting though, both last week and previously, that this is no longer worth fixing.  To that end, this PR adds a simple, javascript-free/CSS only button that can be used by forms.  The primary downside is some (hopefully minimal) increased maintenance overhead on the Bolt side when making updates that affect `<bolt-button>` markup.  This will be accompanied by an update to the `pega_bolt` module to use this new twig template.

@krlucas suggested in our meeting about this last Thursday that we could potentially bring back `<bolt-button>` in forms once we had automated testing coverage.  I'm frankly skeptical, partly because even if we know _that_ form submission is broken, we don't have a clear path to fixing it (which is our current scenario).  I'd love to be proven wrong though.

## How to test

View the Bolt form demo on this feature branch at `pattern-lab/patterns/02-components-form-form-full-campaign-landing/02-components-form-form-full-campaign-landing.html` and confirm parity with the existing form button.
